### DISCUSSION TITLE
Welcome to MapsModelsImporter Discussions!

### DISCUSSION THREAD

**Discussion by eliemichel** [eliemichel/MapsModelsImporter#122] [General]
## 👋 Welcome!
  We’re trying the new "Discussions" feature of GitHub as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

I don't know yet what this new feature brings exactly, feel free to ask things here then and we'll figure out!

**Comment by Virtuelconcept** [comment]
[error_mi.zip](https://github.com/eliemichel/MapsModelsImporter/files/5849740/error_mi.zip)
Hi Elie,
I get some errors using your importer in blender, and i have already tried many versions and all i can but still doesn't work.
Actually i'm using Blender 2.91, Renderdoc 1.10, and the version 3.3 of your importer. 
I send you my console log (errors) and the rdc file.
Hope you can help !!!

**Reply by eliemichel** [reply]
Mmh most importantly you must use the last version of this add-on 0.3.5. Versions for other softwares are consequences (Rd 1.10 and Blender 2.91 are good).

**Reply by adam-bielanski** [reply]
@eliemichel  - I have the sane problem, despite installing  3.5 version - tesed on your sample RDC files too.
"Python: Traceback (most recent call last):
  File "C:\Users\Adam\AppData\Roaming\Blender Foundation\Blender\2.91\scripts\addons\MapsModelsImporter\operators.py", line 54, in execute
    importCapture(context, self.filepath, self.max_blocks, pref)
  File "C:\Users\Adam\AppData\Roaming\Blender Foundation\Blender\2.91\scripts\addons\MapsModelsImporter\google_maps.py", line 305, in importCapture
    captureToFiles(context, filepath, prefix, max_blocks)
  File "C:\Users\Adam\AppData\Roaming\Blender Foundation\Blender\2.91\scripts\addons\MapsModelsImporter\google_maps.py", line 77, in captureToFiles
    out = subprocess.check_output([python, SCRIPT_PATH, filepath, prefix, str(max_blocks)], stderr=subprocess.STDOUT)
  File "C:\Program Files\Blender Foundation\Blender 2.91\2.91\python\lib\subprocess.py", line 411, in check_output
    **kwargs).stdout
  File "C:\Program Files\Blender Foundation\Blender 2.91\2.91\python\lib\subprocess.py", line 488, in run
    with Popen(*popenargs, **kwargs) as process:
  File "C:\Program Files\Blender Foundation\Blender 2.91\2.91\python\lib\subprocess.py", line 800, in __init__
    restore_signals, start_new_session)
  File "C:\Program Files\Blender Foundation\Blender 2.91\2.91\python\lib\subprocess.py", line 1207, in _execute_child
    startupinfo)
OSError: [WinError 87] Parameter is incorrect
location: <unknown location>:-1"

**Reply by eliemichel** [reply]
Do you have `python.exe` in `C:\Program Files\Blender Foundation\Blender 2.91\2.91\python\bin`?
Then can you go to a Python console panel in blender and type this:

```python
import sys
sys.executable
```

and let me know what the output is?

**Comment by Jvincentrs4** [comment]
Elie, 

I was wondering do you have a suggestion for getting this up and running using chrome 64bit. I have tried to install the 32bit version of chrome several times but it always defaults to the 64bit version. Any help would be much appreciated.

Best, J

**Reply by eliemichel** [reply]
I use the 64bit version when running MS Edge with no problem.

**Comment by Mitch9x** [comment]
Hello, are there any plans for an update? Both the model importer and texture packer don´t work in blender 2.93 Lts, while the importer works in 2.83 Lts but the capture merger crashes to desktop in that version.

**Reply by eliemichel** [reply]
I just updated both of them today!

**Reply by Mitch9x** [reply]
awesome news, will try them out!

**Comment by julesfelix** [comment]
Hi Elie,
I was trying your tool and it is really fantastic. 
Because I know Cinema 4d better than Blender I was trying to export the model and I have some issues with the textures or UV's.
Have you tried to use your models in other softeware? Somehow I'm really struggling. Thank you in advance.

**Reply by eliemichel** [reply]
Yes I use simple OBJ files when exporting and it saves all there is to know: mesh, UVs and texture assignment. What did you try so far? Which errors/issues did you experience with your approach?

**Reply by julesfelix** [reply]
Hi Elie, thank you for your reply! I have tried OBJ, COLLADA, FBX and alembic. Also I tried to run it in c4d and 3dmax. The mesh looks fine and with 3dsmax and COLLADA the texture came colorful from the start, allocated at the right points. With the other formats and in c4d I had to relink the png, but the results were the same. The problem is that in both programs, the texture looks very blurry. It's more or less at the right place (UV) - the river is blueish, the roofes are red. But it's very blurry and there is a tiling visible. It does look a bit different in c4d and 3dmax, 3dmax having a bit more detail but also more square tiling artefacts. I have merged the data into a single big mesh in blender. Could that be the problem?

**Reply by eliemichel** [reply]
You can still try without merging to compare, c4d/max might activate a simplified look in the viewer for more fluent interaction, did you check that the issue also appears in rendered images? Could you attach a few screenshot comparing Blender and C4D viewports?

**Reply by julesfelix** [reply]
Hi Elie, you were right. It was just the preview. I never had this behavior in cinema, maybe it has to do with the huge map of the texture, it just looked so sharp in blender that i did not even go on. It worked well with obj, I just had to relink the png. Anyway, thank you so much for your help and this great tool!

**Comment by Craiggrass31** [comment]
Hello, 
I look forward to using this solution! I'm running into an issue and not sure if it's because have windows 11. The Chrome Shortcut changes to Command Prompt but, when I double click it says "Windows cannot find 'C:\Program Files\Google\Chrome\Application\chrome.exe'. Make sure you typed the name correctly, and then try again." Can anyone help? Should this work in windows 11? Thanks for any help.

**Reply by vimaxus** [reply]
Hello,

I'm trying to import a .rdc from geoportail.lu and it almost works. It created the folder and the textures seem there and corectly extracted.
The first valid index is 86 though so I think that is generating the error. I have a debug info output. I can rename the files so that it starts with 0 like the google maps files but I can't trigger only the folder read nor can I repack the "cleaned" folder to a rdc.

Any suggestions?

**Reply by eliemichel** [reply]
@Craiggrass31 You must adapt the custom link so that this part of the text is the original shortcut (the path to where Chrome is installed on your system)
@vimaxus You may quite easily adapt the source code of `google_maps.py` to start at a given index rather than 0.

**Comment by RR2pao** [comment]
Hi Elie,
I have an issue with importing the rdc file, could you please help me? I am using the latest version of your importer (0.4.0) and Blender 3.1.2.

[my rdc file](https://drive.google.com/drive/folders/1pRkgeuKwkPCbNo95NI5XdCSR4zGP6QDf?usp=sharing)
 
![kép](https://user-images.githubusercontent.com/67505133/163805729-4f636f26-d9c8-4237-a291-51c8336627aa.png)

Thank you in advance!

**Reply by eliemichel** [reply]
I guess it does not like the accents in your username, try loading from a rdc file located in a file and directory with no special character (e.g. `C:\Temp\capture.rdc`)

**Comment by Monotung** [comment]
Hi Elie, 

I have problem on loading in the RDC file to blender. there was 3 folder created after loading. Please see the Toggle System Console at the bottom.

Here is my setting.
Chrome latest
Blender 2.90.1
MapModelsImporter 0.3.3
RenderDoc 1.10.64

This setting works last year on my computer. but the hardisk crashed and i am now reinstalled everything but it doesnt work. 

so what i can do to fix it? thank you!!!

Read prefs: C:\Users\User\AppData\Roaming\Blender Foundation\Blender\2.90\config\userpref.blend
found bundled python: C:\Program Files\Blender Foundation\Blender 2.90\2.90\python
google_maps_rd returned:
Loading capture from C:\Users\User\Documents\test 5.rdc...
Trying scraping strategy #4...
Not relevant yet: CopySubresourceRegion(ResourceId::1145, ResourceId::9760)
(Skipping drawcall CopySubresourceRegion(ResourceId::7261, ResourceId::9762))
(Skipping drawcall CopySubresourceRegion(ResourceId::151, ResourceId::9765))
(Skipping drawcall Draw(4))
(Skipping drawcall Draw(4))
(Skipping drawcall Draw(4))
(Skipping drawcall Draw(4))
(Skipping drawcall Draw(4))
(Skipping drawcall Draw(4))
(Skipping drawcall Draw(4))
(Skipping drawcall Draw(4))
(Skipping drawcall Draw(4))
(Skipping drawcall Draw(4))
(Skipping drawcall Draw(4))
(Skipping drawcall Draw(4))
(Skipping drawcall Draw(4))
(Skipping drawcall Draw(4))
(Skipping drawcall Draw(4))
(Skipping drawcall Draw(4))
(Skipping drawcall Draw(4))
(Skipping drawcall Present(ResourceId::164))
Not relevant yet: CopySubresourceRegion(ResourceId::1145, ResourceId::9760)
Not relevant yet: CopySubresourceRegion(ResourceId::7261, ResourceId::9762)
Not relevant yet: CopySubresourceRegion(ResourceId::151, ResourceId::9765)
Not relevant yet: Draw(4)
Not relevant yet: Draw(4)
Not relevant yet: Draw(4)
Not relevant yet: Draw(4)
Not relevant yet: Draw(4)
Not relevant yet: Draw(4)
Not relevant yet: Draw(4)
Not relevant yet: Draw(4)
Not relevant yet: Draw(4)
Not relevant yet: Draw(4)
Not relevant yet: Draw(4)
Not relevant yet: Draw(4)
Not relevant yet: Draw(4)
Not relevant yet: Draw(4)
Not relevant yet: Draw(4)
Not relevant yet: Draw(4)
Not relevant yet: Draw(4)
Not relevant yet: Present(ResourceId::164)
Trying scraping strategy #5...
Not relevant yet: CopySubresourceRegion(ResourceId::1145, ResourceId::9760)
Not relevant yet: CopySubresourceRegion(ResourceId::7261, ResourceId::9762)
Not relevant yet: CopySubresourceRegion(ResourceId::151, ResourceId::9765)
Not relevant yet: Draw(4)
Not relevant yet: Draw(4)
Not relevant yet: Draw(4)
Not relevant yet: Draw(4)
Not relevant yet: Draw(4)
Not relevant yet: Draw(4)
Not relevant yet: Draw(4)
Not relevant yet: Draw(4)
Not relevant yet: Draw(4)
Not relevant yet: Draw(4)
Not relevant yet: Draw(4)
Not relevant yet: Draw(4)
Not relevant yet: Draw(4)
Not relevant yet: Draw(4)
Not relevant yet: Draw(4)
Not relevant yet: Draw(4)
Not relevant yet: Draw(4)
Not relevant yet: Present(ResourceId::164)
Trying scraping strategy #6...
Not relevant yet: CopySubresourceRegion(ResourceId::1145, ResourceId::9760)
Not relevant yet: CopySubresourceRegion(ResourceId::7261, ResourceId::9762)
Not relevant yet: CopySubresourceRegion(ResourceId::151, ResourceId::9765)
Not relevant yet: Draw(4)
Not relevant yet: Draw(4)
Not relevant yet: Draw(4)
Not relevant yet: Draw(4)
Not relevant yet: Draw(4)
Not relevant yet: Draw(4)
Not relevant yet: Draw(4)
Not relevant yet: Draw(4)
Not relevant yet: Draw(4)
Not relevant yet: Draw(4)
Not relevant yet: Draw(4)
Not relevant yet: Draw(4)
Not relevant yet: Draw(4)
Not relevant yet: Draw(4)
Not relevant yet: Draw(4)
Not relevant yet: Draw(4)
Not relevant yet: Draw(4)
Not relevant yet: Present(ResourceId::164)
Trying scraping strategy #7...
Not relevant yet: CopySubresourceRegion(ResourceId::1145, ResourceId::9760)
Not relevant yet: CopySubresourceRegion(ResourceId::7261, ResourceId::9762)
Not relevant yet: CopySubresourceRegion(ResourceId::151, ResourceId::9765)
Not relevant yet: Draw(4)
Not relevant yet: Draw(4)
Not relevant yet: Draw(4)
Not relevant yet: Draw(4)
Not relevant yet: Draw(4)
Not relevant yet: Draw(4)
Not relevant yet: Draw(4)
Not relevant yet: Draw(4)
Not relevant yet: Draw(4)
Not relevant yet: Draw(4)
Not relevant yet: Draw(4)
Not relevant yet: Draw(4)
Not relevant yet: Draw(4)
Not relevant yet: Draw(4)
Not relevant yet: Draw(4)
Not relevant yet: Draw(4)
Not relevant yet: Draw(4)
Not relevant yet: Present(ResourceId::164)
Trying scraping strategy #8...
Not relevant yet: CopySubresourceRegion(ResourceId::1145, ResourceId::9760)
(Skipping drawcall CopySubresourceRegion(ResourceId::7261, ResourceId::9762))
(Skipping drawcall CopySubresourceRegion(ResourceId::151, ResourceId::9765))
(Skipping drawcall Draw(4))
(Skipping drawcall Draw(4))
(Skipping drawcall Draw(4))
(Skipping drawcall Draw(4))
(Skipping drawcall Draw(4))
(Skipping drawcall Draw(4))
(Skipping drawcall Draw(4))
(Skipping drawcall Draw(4))
(Skipping drawcall Draw(4))
(Skipping drawcall Draw(4))
(Skipping drawcall Draw(4))
(Skipping drawcall Draw(4))
(Skipping drawcall Draw(4))
(Skipping drawcall Draw(4))
(Skipping drawcall Draw(4))
(Skipping drawcall Draw(4))
(Skipping drawcall Draw(4))
(Skipping drawcall Present(ResourceId::164))
Error: Could not find the beginning of the relevant 3D draw calls
Scraping capture from none...

Info: Deleted 3 object(s)

**Reply by eliemichel** [reply]
All draw calls have only 4 indices, this is a hint that your capture contains no 3D data, you likely did not turn on 3D in Google Maps or did not move in the viewport while capturing.

**Comment by adrianfx360** [comment]
Hola!!! intento importar pero me salta esto?:
Hello!! Does anyone know how I could solve this?:

Python: Traceback (most recent call last):
  File "C:\Users\Usuario\AppData\Roaming\Blender Foundation\Blender\3.1\scripts\addons\MapsModelsImporter\operators.py", line 53, in execute
    try:
  File "C:\Users\Usuario\AppData\Roaming\Blender Foundation\Blender\3.1\scripts\addons\MapsModelsImporter\google_maps.py", line 194, in importCapture
    
  File "C:\Users\Usuario\AppData\Roaming\Blender Foundation\Blender\3.1\scripts\addons\MapsModelsImporter\google_maps.py", line 36, in captureToFiles
    First turn on debug output by activating the "Debug Info"-checkbox under Edit > Preferences > Add-ons > MapsModelsImporter
IndexError: list index out of range

location: <unknown location>:-1

**Reply by eliemichel** [reply]
From the error message you attach: "First turn on debug output by activating the "Debug Info"-checkbox under Edit > Preferences > Add-ons > MapsModelsImporter"

**Comment by DavixDevelop** [comment]
Hi, I just have one question regarding the scale. What value would I need to set it for the model to be IRL scale. I'm asking because I want to place the whole capture on a oblate spheroid, but since I don't know the the correct scale, I can't can't create a oblate spheroid with the correct radius of the Earth, and place the capture on it. I could try to manually scale, rotate and shift the capture, but I want for things to be as accurate as possible. BTW, I really love this addon. Great work.

**Reply by eliemichel** [reply]
Hi, unfortunately units are a known limitation of this process, because the actual physical units are present nowhere in the data we extract from Google Maps, and I was unable to figure out the inconsistency across various captures.

One solution is to measure a known distance both in Blender (`distance_blender`) and on the map (`distance_physical`) then compute `scale = distance_physical / distance_blender` and finally scale the whole scene by `scale`. **Pro-tip** Once you start a scaling operation with the 'S' key, you can directly input an exact scaling value using the keyboard.

**Comment by catalara99** [comment]
Hello everyone! I’ve been trying for 3 days to get this wonderful tool to work, but I’m not succeeding at all! I would like to solve my problem, but my time is running out and I really need to export a 3D image of a specific area from Google Maps to Sketchup.

I watched a lot of videos, read and re-read these forums, tested all versions of MapsModelsImporter, Blender and RenderDoc. But I was unsuccessful! Can anyone generate this map for me? It would be essential for my course completion work.

The area of my interest is in Vila Velha, Espírito Santo, Brazil. I will leave a print with the desired cut highlighting the points of interest. I will be eternally grateful to anyone who can help me, capturing this 3D map with textures and sending it to Blender and exporting it to a format compatible with Sketchup

![Screenshot_2](https://github.com/eliemichel/MapsModelsImporter/assets/156985181/67151617-c103-4873-bca6-3f2b6248509e)

https://www.google.com.br/maps/@-20.321665,-40.2928094,5307m/data=!3m1!1e3?entry=ttu

**Reply by catalara99** [reply]
contact: +5527999212271
talaricopt@gmail.com

**Comment by Navicula** [comment]
Map Models Importer is not showing up i N-Panel
MacBook Pro M4 OS15.4.1
Blender 4.3.2
Map Models Importer 0.7.0
MMI is imported to Preferences / Add On and has a check mark, but dosent show up in N-Panel Layout / Object Mode / N- SIdepanel, but can be seen in N-Panel / Tool / Workspace, together with active Add-Ons that shows up in the sidepanel.
